File: 剑指offer/58.1.翻转单词顺序.py
class Solution:
    def reverseWords(self, s: str) -> str:
        s = s.strip()

        i = j = len(s) - 1
        words = []
        while i >= 0:
            while i >= 0 and s[i] != ' ':
                i -= 1
            words.append(s[i + 1:j + 1])

            while i >= 0 and s[i] == ' ':
                i -= 1

File: 剑指offer/58.1.翻转单词顺序.cpp
#include <string>
#include <vector>

using namespace std;

class Solution {
public:
    string reverseWords(string s)
    {
        vector<string> words;
        string word = "";
        for (char c : s) {
            if (c == ' ') {
                if (word != "") {
                    words.push_back(word);
                    word = "";
                }
            } else {
                word.push_back(c);
            }
        }

        if (word != "") {
            words.push_back(word);
            word = "";
        }

        string ans = "";

        for (int i = words.size() - 1; i >= 0; i--) {
            ans += words[i];
            ans += " ";
        }

        ans.pop_back();

        return ans;
    }
};